### PR TITLE
Reuse master

### PR DESCRIPTION
**Pull request by profxj**
Offers a new approach to reuse MasterFrames: 
  Loads from hard drive after the first set are done.

This would likely supersede the reuseMaster flag in armsld.

Added  tests for armbase (had to turn off the arcid_plot test)

Failing Python3 because there is no xrange (its range actually behaves
like Python2's xrange).

Passed all test suite redux

### CODE REVIEW

**Comment by rcooke-ast** [comment]
I think we've spoken about this before, but I can't seem to find the email...  The original intention of storing the master frames is to avoid remaking the master frames that use identical input frames. The plan is:

(1) If a master frame composed of images IM1, IM2, IM3... hasn't previously been made, make it.
(2) Store this master frame in the masters index, i
(3) For all science frames that use this master frame, store the index, i, in arsciexp.

This avoids read/write overhead, and avoids storing repeated master frames in arsciexp. So, I think we need to avoid storing unneeded frames in arsciexp (i.e. the slf class that is carried throughout the reduction).

**Comment by rcooke-ast** [comment]
Call me a dunce (!!), but I can't see what lines of code write/read the masters to/from disk. Sorry for being slow, but could you point it out to me, please  :-)

**Comment by profxj** [comment]
The write statement is 194 of armlsd.py in this branch:
            armasters.save_masters(slf, det, setup)

The read statements are sprinkled throughout the code.  Mainly
in arsciexp and arlmsd

While I agree that excessive I/O is bad, I think this is rather cheap
compared to making a full memory copy of each file for every science frame.
That could climb to a few Gb rather quickly..

Also, I wasn't too keen on coding up the bits and pieces for slf._lordloc, slf._pixcen, etc.
which are not in MasterFrames (thus far).

File: pypit/arutils.py [hunk 1]
```diff
@@ -133,27 +133,66 @@ def calc_offset(raA, decA, raB, decB, distance=False):
...
+
+    """
+    fitsdict = {}
+    fitsdict['date'] = ['2015-01-23T00:54:17.04']*nfile
+    fitsdict['target'] = ['Dummy']*nfile
+    #
+    return fitsdict
+
+
+def dummy_self(pypitdir=None, inum=0, fitsdict=None):
```
Comment (rcooke-ast) [inline pypit/arutils.py]: Should we also specify `nfile=10` here, as one of the arguments?

and then replace line 176 with `spect[key]['index'] = [[jj]*nfile]`
and replace line 181 with `fitsdict = dummy_fitsdict(nfile=nfile)`

Comment (profxj) [inline pypit/arutils.py]: Good call.  Done.

**Comment by profxj** [comment]
Merging.

**Pull request merged**